pkg/scheduler/algorithm/predicates/csi_volume_predicate.go

```diff
@@ -70,7 +73,7 @@ func (c *CSIMaxVolumeLimitChecker) attachableLimitPredicate(
 
 	// a map of unique volume name/csi volume handle and volume limit key
 	newVolumes := make(map[string]string)
-	if err := c.filterAttachableVolumes(pod.Spec.Volumes, pod.Namespace, newVolumes); err != nil {
+	if err := c.filterAttachableVolumes(nodeInfo, pod.Spec.Volumes, pod.Namespace, newVolumes); err != nil {
```

Comment:
We're missing a check (probably somewhere around L108) that limit should == 0 if an entry doesn't exist.

We may be able to make future optimizations around this area too to not have to count volumes if no CSI drivers are installed on the node.

Reply:
Actually there's two cases we need to handle:
- csi driver not installed: limit is 0
- csi driver installed but doesn't report limits: limit is infinite

Reply:
Can you take a look if https://github.com/kubernetes/kubernetes/pull/77595/commits/e7137eaa9b5795e4e4da6cd9c93014eebf993711 addresses this?

I'm assuming that a `0` limit means no limit. The specs says that the CO should decide how many volumes can be published if the driver reports `.

Reply:
Sorry, new commit: https://github.com/kubernetes/kubernetes/pull/77595/commits/07651ea160b09b7f9637acb25cb145ce41dfa726

Reply:
CC @gnufied

Reply:
This should be fixed now.